src/components/NavBar/utils.ts

```diff
@@ -29,13 +30,14 @@ type LayersCategoriesType = LayersCategoryType[];
 
 type MenuItemsType = MenuItemType[];
 
+// TODO better typing for layer key
```

Comment:
Somewhat. In line 35 I was hoping that the type of `layersList` could be `[key: string]: LayerKey[]` but this brought issues as TS can only verify JSON keys, meaning we would be moving the casting to wherever this function is used instead of at line 40. 

What do you think? Should I just remove this comment or cast where this function is used?

Reply:
Hrm. Where _is_ this function used? Generally speaking, I think it's best to assign the best types available as "early" in the app possible. In this case, that would mean that whatever entities get used as input args for this function, get a `[key: string]: LayerKey[]` type as early as possible on import. You could also use a guard function (`isLayerKey`)...

Reply:
Should have linked it prior haha: [utils.ts](https://github.com/WFP-VAM/prism-frontend/blob/d2715adbec436163ea9bcb914efaa5a9ac8da0d6/src/components/NavBar/utils.ts#L53)

I like that ideology, I'll type it here then and remove the comment. I'll also make a quick guard function with `val is LayerKey` so TS will be happy about passing it into this function :+1: